File: frontend/src/types/commands.ts
// frontend/src/types/commands.ts

export interface CommandRequest {
  domain: string;
  command: string;
  params?: Record<string, unknown>;
}

export interface CommandResponse {
  success: boolean;
  data?: unknown;
  error?: string;
  duration_ms?: number;
}

export interface CapabilityDomain {
  name: string;
  description: string;
  version: string;
  commands: CapabilityCommand[];
}

export interface CapabilityCommand {
  name: string;
  description: string;
  params?: CapabilityParam[];
}

export interface CapabilityParam {
  name: string;
  type: "string" | "integer" | "number" | "boolean" | "array" | "object";
  description?: string;
  required?: boolean;
  default?: unknown;
  enum?: string[];
}

export interface CommandHistoryEntry {
  id: string;
  domain: string;
  command: string;
  params?: Record<string, unknown>;
  response: CommandResponse;
  timestamp: number;
  isFavorite: boolean;
}

export interface Settings {
  model: string;
  effort: string;
  workflow: string;
  access_mode: string;
  directive: string;
  max_sessions: number;
}

File: src/cortex_webkit/models/commands.py
# src/cortex_webkit/models/commands.py
"""Command request/response models."""

from typing import Any
from pydantic import BaseModel


class CommandRequest(BaseModel):
    domain: str
    command: str
    params: dict[str, Any] | None = None


class CommandResponse(BaseModel):
    success: bool
    data: Any = None
    error: str | None = None
    duration_ms: int | None = None


class StatusResponse(BaseModel):
    connected: bool
    port: int | None = None
    pid: int | None = None
    project: str | None = None
    domains: list[str] | None = None


class SettingsResponse(BaseModel):
    model: str
    effort: str
    workflow: str
    access_mode: str
    directive: str
    max_sessions: int


class SettingsUpdate(BaseModel):
    model: str | None = None
    effort: str | None = None
    workflow: str | None = None
    access_mode: str | None = None
    directive: str | None = None
